test/integration/testpaths/testpath_volume_cuncurrent_snapshots.py

```diff
@@ -0,0 +1,827 @@
+            cls.service_offering = ServiceOffering.create(
+                cls.apiclient,
+                cls.testdata["service_offering"],
+            )
+            cls._cleanup.append(cls.service_offering)
+
+            cls.vm_pool = []
+            for i in range(4):
+                cls.vm = VirtualMachine.create(
+                    cls.apiclient,
+                    cls.testdata["small"],
+                    templateid=cls.template.id,
+                    accountid=cls.account.name,
+                    domainid=cls.account.domainid,
+                    serviceofferingid=cls.service_offering.id,
+                    zoneid=cls.zone.id,
+                    mode=cls.zone.networktype
+                )
+
+                cls.vm_pool.append(cls.vm)
+                cls._cleanup.append(cls.vm)
+
+            cls.chekcsum_pool = []
+            cls.root_pool = []
+            cls.snapshot_pool = []
+            cls.rec_policy_pool = []
+            for vm in cls.vm_pool:
+                root_volumes = list_volumes(
+                    cls.apiclient,
+                    virtualmachineid=vm.id,
+                    type='ROOT',
+                    listall=True
+                )
+
+                checksum_root = createChecksum(
+                    cls.testdata,
+                    vm,
+                    root_volumes[0],
+                    "rootdiskdevice")
+
+                cls.chekcsum_pool.append(checksum_root)
+                cls.root_pool.append(root_volumes[0])
+            try:
+                cls.pools = StoragePool.list(cls.apiclient, zoneid=cls.zone.id)
+            except Exception as e:
+                raise unittest.SkipTest(e)
+
+        except Exception as e:
+            cls.tearDownClass()
+            raise e
+        return
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            for vm in cls.vm_pool:
+                vm.delete(cls.apiclient)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+    def setUp(self):
+        self.apiclient = self.testClient.getApiClient()
+        self.dbclient = self.testClient.getDbConnection()
+        self.cleanup = []
+
+    def tearDown(self):
+        try:
+            cleanup_resources(self.apiclient, self.cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    @classmethod
+    def RestartServer(cls):
+        """Restart management server"""
+
+        sshClient = SshClient(
+            cls.mgtSvrDetails["mgtSvrIp"],
+            22,
+            cls.mgtSvrDetails["user"],
+            cls.mgtSvrDetails["passwd"]
+        )
+        command = "service cloudstack-management restart"
+        sshClient.execute(command)
+
+        return
+    @classmethod
+    def StopVM(cls, vms):
+        for vm in vms:
+            vm.stop(cls.apiclient)
+        return
+
+    @attr(tags=["advanced", "basic"])
```

Comment:
Add required_hardware tag. Set value to false as this test case can't be run on simulator